norddrop/src/device/types.rs

```diff
@@ -1,65 +1,30 @@
 use std::time::SystemTime;
 
-use drop_transfer::{utils::Hidden, File as _, FileId, Transfer};
+use drop_transfer::{utils::Hidden, File as _, Transfer};
 use serde::{Deserialize, Serialize};
-use uuid::Uuid;
 
-use super::utils;
-
-#[derive(Deserialize, Debug)]
+#[derive(Serialize, Deserialize, Debug)]
 pub struct TransferDescriptor {
     pub path: Hidden<String>,
     pub content_uri: Option<url::Url>,
     pub fd: Option<i32>,
 }
 
-#[derive(Serialize)]
-pub struct EventTransferRequest {
-    peer: String,
-    transfer: String,
-    files: Vec<File>,
-    timestamp: u64,
-}
-
-#[derive(Serialize)]
-pub struct EventRequestQueued {
-    peer: String,
-    transfer: String,
-    files: Vec<File>,
-    timestamp: u64,
-}
-
-#[derive(Serialize)]
-struct File {
-    id: String,
-    path: String,
-    size: u64,
-}
-
-#[derive(Serialize)]
-pub struct StartEvent {
-    transfer: String,
-    file: String,
-    transfered: u64,
-    timestamp: u64,
-}
-
-#[derive(Serialize)]
-pub struct ProgressEvent {
-    transfer: String,
-    file: String,
-    transfered: u64,
-    timestamp: u64,
+#[derive(Serialize, Deserialize)]
+pub struct File {
+    pub id: String,
+    pub path: String,
+    pub size: u64,
 }
 
-#[derive(Serialize)]
+#[derive(Serialize, Deserialize)]
 pub struct Status {
-    status: u32,
+    pub status: u32,
     #[serde(skip_serializing_if = "Option::is_none")]
-    os_error_code: Option<i32>,
+    pub os_error_code: Option<i32>,
 }
 
-#[derive(Serialize)]
+#[derive(Serialize, Deserialize)]
```

Comment:
All of the documentation about the types you can find here:
https://mozilla.github.io/uniffi-rs/Overview.html

For a slightly quicker way of glancing at what is possible to be used without any internal details, you could look at the tables here
https://mozilla.github.io/uniffi-rs/internals/lifting_and_lowering.html

Reply:
@LukasPukenis I'm forwarding the uniffi calls into the `Device` methods. These use JSON. I'm doing this to not break the swig bindings for now as it would explode this PR size even bigger.
I'll clean this up in the subsequent PRs

Reply:
Greatly appreciated!